In [3]:
import os
os.environ['CUDA_VISIBLE_DEVICES'] = '-1'

In [7]:
import pandas as pd
import keras

df = pd.read_csv("EEG data - Sheet1.csv")
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Columns: 322 entries, Unnamed: 0 to target
dtypes: float64(320), int64(2)
memory usage: 100.8 KB


# Binary classification using SVM
our data has much more features than samples `RangeIndex: 40 entries, 0 to 39 Columns: 322 entries`
`If the number of features is much greater than the number of samples, avoid over-fitting in choosing Kernel functions and regularization term is crucial.`
## Two main ideas of SVM
svm is more like an optimization for linear classification
- Find the maximal margin. Draw tree lines: one to capture data points of one class, one to capture data points of another class, last one to measure their distance. The two first lines are also the border of each class. How the border is defined? Visualy, it's kind evident, but why we know it's a border?? two kinds of margin: hard and soft
- Sometimes, data points are not separable in the current dimension, so we can transform our data in higher dimension to find the separable border(why not lower?). And this transformation can be expensive, so we have different ways to do this transformation, which are also called kernels. For example: `polynominal kernel`, `radial basis function rbf`, `sigmoid kernel`.

In [9]:
from sklearn import svm
X = [[0, 0], [1, 1]]
y = [0, 1]
clf = svm.SVC()
clf.fit(X, y)
clf.predict([[2., 2.]])

array([1])